### DISCUSSION TITLE
Question about Object layer positioning (Isometric)

### DISCUSSION THREAD

**Discussion by lpturmel** [adrien-bon/bevy_ecs_tiled#25] [Q&A] [answered]
Hi!

First of all, thank you for this amazing library—it's a really cool project. I'm relatively new to Bevy game development, especially when it comes to working with isometric layouts, so it's possible I'm missing something obvious.

I’m reaching out for guidance on troubleshooting an issue I’m facing with object layers registered through `register_tiled_object`. The custom objects I load don’t appear at the same position in Bevy as they do in Tiled. I’m trying to figure out whether this could be caused by something in my Tiled file (e.g., incorrect dimensions or settings), or if it’s more related to my code.

### Map setup:

```rust
fn setup_tilemap(mut commands: Commands, asset_server: Res<AssetServer>) {
    let map_handle: Handle<TiledMap> = asset_server.load("level0.tmx");
    commands.spawn((
        TiledMapBundle {
            tiled_map: map_handle,
            render_settings: TilemapRenderSettings {
                render_chunk_size: UVec2::new(1, 1),
                y_sort: true,
            },
            tiled_settings: TiledMapSettings {
                map_positioning: MapPositioning::Centered,
                collider_callback: |entity_commands| {
                    entity_commands.insert(RigidBody::Static);
                },
                ..default()
            },
            ..Default::default()
        },
        Level,
    ));
}
```

### Object registration:
```rust
pub struct TerrainPlugin;

impl Plugin for TerrainPlugin {
    fn build(&self, app: &mut App) {
        app
            .register_tiled_object::<TileWallObj>("wall");
    }
}

#[derive(TiledObject, Component, Default, Reflect, Debug)]
pub struct TileWallObj {}
```

### Expected in Tiled:
![image](https://github.com/user-attachments/assets/29dcb1bc-a931-48e1-a3bd-f8138c8be546)

### Inside of Bevy
![image](https://github.com/user-attachments/assets/a40ad094-a18c-417e-853f-b664f008af5c)

Is this working correctly and I need to like do some equation to get back the position I'm expecting? If so, is there an example somewhere that could give me an idea of the implementation.

Thank you!

**Comment by adrien-bon** [comment, marked as answer]
Hey!

Sorry, never saw the discussion until I got the notification you closed it...
Still a bit new to Gihub!

From what I understand you have an issue with Tiled object not being at the correct position for isometric maps.
Indeed, it was quite buggy.

If you are using "diamond" isometric map, it should be fixed in `dev` branch, following [this commit](https://github.com/adrien-bon/bevy_ecs_tiled/commit/839110b0dbe1748eb6e827c16e149dc3b6f9435e), although there are still issues regarding physics colliders.
If you are using "staggered" isometric map, unfortunately there is still an issue and I choose not to fix it right now and keep it as a limitation of the crate (there should be a warning message).

Please, let me know if it works for you!
Also, if you encounter any issue, feel free to open an issue.

Thanks!

**Reply by lpturmel** [reply]
Hi!

No problem at all this was more of an experimentation project on my side.

Thanks for the info, for now I moved my project to orthogonal tiles and everything works pretty much as expected.

I'll keep this in mind for now if I ever work in isometric in the future.

Thank you!
